### DISCUSSION TITLE
Use external and requires in base type

### DISCUSSION THREAD

**Discussion by hchen** [ExpediaGroup/graphql-kotlin#1565] [Q&A] [answered]
Hi team, we are getting the following errors when trying to use  `@ExternalDirective` and `@RequiresDirective` in base type.

```
com.expediagroup.graphql.generator.federation.exception.InvalidFederatedSchema: Invalid federated schema:
 - base Product type has fields marked with @requires directive, validatedField=shippingEstimate
 - base Product type has fields marked with @external directive, fields=[weight]
```

We want to caculate `shippingEstimate` of the `Product` which relies on the `weight` field. This weight field is resolved by another subgraph.

I'm wondering why these checks are added and if this can be disabled using a config.

Thanks in advance for your help.

**Comment by dariuszkuc** [comment, marked as answer]
In general, the expectation was that you should not define `@external` fields on the base type -> given a type you would extend its functionality in other subgraphs. In your case, you would compute `shippingEstimate` in the subgraph that calculates the `weight`. I think that is generally an anti-pattern.

That being said, I think that would still be a valid graph for the composition so we probably should remove this check. Also with the smart type merging introduced in Fed v2, `@extends / extend type` is no longer necessary so it might be hard to distinguish between base and extended type anyway.
